test/e2e/fqdn_dns_cache_test.go

```diff
@@ -0,0 +1,318 @@
+	agnHostPodTwoIP, _ := agnHostPodIps[1].AsStrings()
+
+	// create customDNS service and get its ClusterIP.
+	customDnsService, err := data.CreateServiceWithAnnotations("custom-dns-service", data.testNamespace, dnsPort,
+		dnsPort, corev1.ProtocolUDP, map[string]string{"app": "custom-dns"}, false,
+		false, corev1.ServiceTypeClusterIP, ptr.To[corev1.IPFamily](corev1.IPv4Protocol), map[string]string{})
+	require.NoError(t, err, "Error creating customDNS service: %+v", err)
+	dnsServiceIP := customDnsService.Spec.ClusterIP
+
+	// create a ConfigMap for the custom DNS server, mapping IP of agnHost pod 1 to the FQDN.
+	dnsConfigData := createDnsConfig(t, map[string]string{"ipAddress": agnHostPodOneIP, "domainName": testFQDN})
+	configMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "custom-dns-config",
+			Namespace: data.testNamespace,
+		},
+		Data: dnsConfigData,
+	}
+	customDnsConfigMap, err := data.CreateConfigMap(configMap)
+	require.NoError(t, err, "failed to create custom dns ConfigMap: %v", err)
+
+	createCustomDnsPod(t, data)
+
+	// set the custom DNS server IP address in Antrea configMap.
+	setDnsServerAddressInAntrea(t, data, dnsServiceIP)
+	defer setDnsServerAddressInAntrea(t, data, "") //reset after the test.
+
+	createFqdnPolicyInNamespace(t, data, testFQDN)
+	createToolboxPod(t, data, dnsServiceIP)
+
+	curlTarget := func(podName, containerName, fqdn string) (string, error) {
+		cmd := []string{"curl", fqdn}
+		stdout, stderr, err := data.RunCommandFromPod(data.testNamespace, podName, containerName, cmd)
+		if err != nil {
+			return "", fmt.Errorf("error when running command '%s' on Pod '%s': %v, stdout: <%v>, stderr: <%v>",
+				strings.Join(cmd, " "), podName, err, stdout, stderr)
+		}
+		return stdout, nil
+	}
+
+	assert.EventuallyWithT(t, func(collect *assert.CollectT) {
+		stdout, err := curlTarget(toolboxPodName, toolboxContainerName, testFQDN)
+		assert.NoError(t, err)
+		t.Logf("response of curl to FQDN - %s", stdout)
+	}, 2*time.Second, 100*time.Millisecond, "trying to curl the fqdn : ", testFQDN)
+
+	// confirm that the FQDN resolves to the expected IP address and store it to simulate caching of this IP associated with the FQDN.
+	t.Logf("Resolving FQDN to simulate caching the current IP inside Toolbox pod...")
+	resolvedIP, err := data.runDNSQuery(toolboxPodName, toolboxContainerName, data.testNamespace, testFQDN, false, dnsServiceIP)
+	fqdnIp := resolvedIP.String()
+	require.NoError(t, err, "failed to resolve FQDN to an IP from toolbox pod : %v", err)
+	require.Equalf(t, agnHostPodOneIP, fqdnIp, "The IP set against the FQDN in the DNS server should be the same, but got %s instead of %s", fqdnIp, agnHostPodOneIP)
+	t.Logf("Successfully received the expected IP %s using the dig command against the FQDN", fqdnIp)
+
+	// update the IP address mapped to the FQDN in the custom DNS ConfigMap.
+	t.Logf("New IP to update to DNS | ipAdress : %s", agnHostPodTwoIP)
```

Comment:
```suggestion
	t.Logf("Updating host mapping in DNS server config to use new IP: %s", agnHostPodTwoIP)
```

I don't understand what are the `|` for in your log messages